pkg/i2gw/notifications/notifications.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifications
+
+import (
+	"fmt"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+)
+
+func init() {
+	CommonNotification = NotificationAggregator{make([]Notification, 0)}
+}
+
+const (
+	InfoNotification    MessageType = "INFO"
+	WarningNotification MessageType = "WARNING"
+	ErrorNotification   MessageType = "ERROR"
+)
+
+type MessageType string
+
+type Notification struct {
+	Type     MessageType
+	Message  string
+	Provider string
+}
+
+type NotificationAggregator struct {
+	Notifications []Notification
```

Comment:
Since we already have the provider name, why not use a `map[string]Notification`, with the providers as the key? This way we'd save the aggregation step we perform in the `ProcessNotifications` function

Reply:
So do you mean something like: 
```go
type NotificationAggregator struct {
	Notifications map[string][]Notification
}
var CommonNotifications NotificationAggregator

CommonNotification.Notifications[string(ProviderName)] = append(CommonNotification.Notifications[string(ProviderName)], newNotification)
```